### PR TITLE
Set all default values

### PR DESCRIPTION
**Pull request by burnout87**
Extend the par_dic var adding the default values for the specific instrument

### CODE REVIEW

**Review by volodymyrss — CHANGES_REQUESTED** [review]
Hm, seems quite simple, maybe it is indeed the natural approach. I am a bit concerned if it will remain natural for the user. It should be. We should verify with plugin test. And also oda_api test.

File: tests/test_job_management.py [hunk 1]
```diff
@@ -1578,7 +1579,12 @@ def test_email_very_long_unbreakable_string(length, dispatcher_long_living_fixtu
 
     assert jdata['exit_status']['email_status'] == 'email sent'
     params['use_scws'] = 'no'
-    products_url = get_expected_products_url(params, 
+    '''
+    included also default values, 
+    which for the case of numerical query, is p, with a value of 10.0
+    and string_like_name
+    '''
+    products_url = get_expected_products_url({**params, 'p': 10.0, 'string_like_name': 'default-name'},
```
Comment (volodymyrss) [inline tests/test_job_management.py:1587]: we could also check in the job directory stored analysis parameters

**Comment by volodymyrss** [comment]
they should be set for given query/product and only for it.

File: tests/tests/reference_emails/submitted-numeric-not-very-long-numerical.html [hunk 2]
```diff
@@ -0,0 +1,71 @@
+<!DOCTYPE html>
```
[the remark this answers is not in the document]
Comment (burnout87) [reply tests/tests/reference_emails/submitted-numeric-not-very-long-numerical.html:1]: yes, but I don't understand why this got commited

File: tests/test_server_basic.py [hunk 3]
```diff
@@ -211,7 +211,7 @@ def test_consistency_parameters_json_dump_file(dispatcher_live_fixture, request_
     jdata = ask(server,
                   params,
                   expected_query_status=["done"],
-                  max_time_s=50,
+                  max_time_s=350,
```
[the remark this answers is not in the document]
Comment (burnout87) [reply tests/test_server_basic.py:214]: no that was just a temporary adjustment that I did while debugging

File: cdci_data_analysis/flask_app/dispatcher_query.py [hunk 4]
```diff
@@ -183,6 +183,7 @@ def __init__(self, app, instrument_name=None, par_dic=None, config=None, data_se
                 logger.debug("NOT get_meta_data request: yes scratch_dir")
 
                 self.set_sentry_client()
+                # TODO is also the case of call_back to handle ?
```
[the remark this answers is not in the document]
Comment (burnout87) [reply cdci_data_analysis/flask_app/dispatcher_query.py:186]: I just double-checked, it does not indeed

File: tests/test_job_management.py (hunk 1 above)
Comment (burnout87) [reply tests/test_job_management.py:1587]: I imagine you mean the scratch_dir right?

Comment (volodymyrss) [reply tests/test_job_management.py:1587]: well yes

**Comment by burnout87** [comment]
> they should be set for given query/product and only for it.

Exactly! It considers only the request query/product default values.

**Comment by burnout87** [comment]
> Hm, seems quite simple, maybe it is indeed the natural approach. I am a bit concerned if it will remain natural for the user. It should be. We should verify with plugin test. And also oda_api test.

I had the same feeling. Definitely needs to be verified also in the plugins and oda_api

**Comment by volodymyrss** [comment]
> > they should be set for given query/product and only for it.
> 
> Exactly! It considers only the request query/product default values.

I mean it'd be good to test that. Is it tested implicitly? Can it be made more explicit?

**Comment by burnout87** [comment]
> > > they should be set for given query/product and only for it.
> > 
> > 
> > Exactly! It considers only the request query/product default values.
> 
> I mean it'd be good to test that. Is it tested implicitly? Can it be made more explicit?

It verifies now that, in case an additional parameter is passed to the request this is found then in returned products, otherwise this is not added since it is not defined for that particular instrument/product/query

**Comment by burnout87** [comment]
> > > they should be set for given query/product and only for it.
> > 
> > 
> > Exactly! It considers only the request query/product default values.
> 
> I mean it'd be good to test that. Is it tested implicitly? Can it be made more explicit?

And also that a parameter that is specific for another instrument ('string_like_name' defined for the instrument 'empty_async') is not set in this case.

File: tests/tests/reference_emails/submitted-numeric-not-very-long-numerical.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (volodymyrss) [reply tests/tests/reference_emails/submitted-numeric-not-very-long-numerical.html:1]: still here!

**Review by volodymyrss — CHANGES_REQUESTED** [review]

**Comment by burnout87** [comment]
They're gone now!

**Review by volodymyrss — APPROVED** [review]
Ok, just one thing, I am a bit perplexed why there are so many new very similar templates now. I used these ignore patterns to exclude and compare separately isolated parts which are changing.
Reference is good since it can be opened and inspected visually for non-obvious rendering problems.
But ok, not a big deal.

**Comment by burnout87** [comment]
> Ok, just one thing, I am a bit perplexed why there are so many new very similar templates now. I used these ignore patterns to exclude and compare separately isolated parts which are changing.
> Reference is good since it can be opened and inspected visually for non-obvious rendering problems.
> But ok, not a big deal.



> Ok, just one thing, I am a bit perplexed why there are so many new very similar templates now. I used these ignore patterns to exclude and compare separately isolated parts which are changing.
> Reference is good since it can be opened and inspected visually for non-obvious rendering problems.
> But ok, not a big deal.

Thanks. In my case I generated a couple new templates since now all the default values will be set in the api_code, and therefore the output api_code for the `empty` instrument will be different from the `empty-async` instrument.

**Pull request merged**